pkg/watch/manager.go

```diff
@@ -246,6 +246,10 @@ func (wm *Manager) replaceWatches(r *Registrar) error {
 		}
 	}
 
+	if err := wm.metrics.reportGvkIntentCount(int64(len(desired))); err != nil {
```

Comment:
We should probably update this metric as part of mutating recordKeeper when adding/removing/replacing watches.

This will help prevent bugs in calling watch manager methods from causing inaccurate reporting of the intended watch count.

Reply:
moved this to part of `recordKeeper`. added a key for registrar name. however, since there is `constrainttemplate-controller-status` registar, it doesn't add up to total gvk.